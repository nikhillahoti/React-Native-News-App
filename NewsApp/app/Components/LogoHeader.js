
import React from 'react';
import {
    View,
    StyleSheet,
    Text
} from 'react-native';
import Logo from 'react-native-vector-icons/FontAwesome';

const LogoHeader = () => {
    return (
        <View style={styles.mainContainer}>
            <View style={styles.container}>
                <Logo name="newspaper-o" color="#ffffff"  size={20} />
                <Text style={styles.header}>News App</Text>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#3B5998',
        width: '100%',
        marginTop: 15,
        textAlign: 'center'
    },
    header: {
        fontSize: 14,
        color: '#ffffff',
        paddingLeft: 20,
        fontWeight:'bold',
        textAlign: 'center'
    },
    mainContainer: {
        height: 50,
        marginTop: 30,
        backgroundColor: '#3B5998'
    }
});

export default LogoHeader;