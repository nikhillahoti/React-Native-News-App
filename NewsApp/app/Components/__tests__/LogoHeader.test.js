import React from 'react';
import renderer from 'react-test-renderer';

import LogoHeader from '../LogoHeader';

describe('Logo Header', () => {

    it('It displays the Logo Header correctly', () => {
        const tree = renderer.create(<LogoHeader />)
                        .toJSON();
        expect(tree).toMatchSnapshot();
    });

});