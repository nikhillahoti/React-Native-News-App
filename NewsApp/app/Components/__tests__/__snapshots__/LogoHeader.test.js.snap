// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Header It displays the Logo Header correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#3B5998",
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "height": 40,
      "marginTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3B5998",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï‡ª
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 14,
          "fontWeight": "bold",
          "paddingLeft": 20,
          "textAlign": "center",
        }
      }
    >
      The Herald
    </Text>
  </View>
</View>
`;
